File: OOP/Car/Ts/Car.ts
interface Vehicle { start(): void }

class Car implements Vehicle {
    private engineId: string;
    constructor(engineId: string) { this.engineId = engineId; }
    start() { console.log(`Car ${this.engineId} starts!`) }
}

const v: Vehicle = new Car("ENG-001");
v.start();

File: OOP/Car/Rust/Car.rs
trait Vehicle { fn start(&self); }

struct Car { engine_id: String }

impl Vehicle for Car {
    fn start(&self) {
        println!("Car {} starts!", self.engine_id);
    }
}

fn main() {
    let v: Box<dyn Vehicle> = Box::new(Car { engine_id: "ENG-001".into() });
    v.start();
}